src/intptrcast.rs

```diff
@@ -21,9 +21,16 @@ pub struct GlobalStateInner {
     /// they do not have an `AllocExtra`.
     /// This is the inverse of `int_to_ptr_map`.
     base_addr: FxHashMap<AllocId, u64>,
+    /// Whether an allocation has been exposed or not. This cannot be put
+    /// into `AllocExtra` for the same reason as `base_addr`.
+    exposed: FxHashSet<AllocId>,
     /// This is used as a memory address when a new pointer is casted to an integer. It
     /// is always larger than any address that was previously made part of a block.
     next_base_addr: u64,
+    /// Whether to allow "permissive provenance" rules. Enabling this means int2ptr casts return
+    /// pointers with "wildcard" provenance that basically match that of all exposed pointers
+    /// (and SB tags, if enabled).
+    permissive_provenance: bool,
     /// Whether to enforce "strict provenance" rules. Enabling this means int2ptr casts return
     /// pointers with an invalid provenance, i.e., not valid for any memory access.
     strict_provenance: bool,
```

Comment:
The comments show quite clearly that permissive and strict provenance are mutually exclusive.

Can you add an enum like this (with comments explaining what the modes do)?
```rust
enum ProvenanceMode { Permissive, Strict, Legacy }
```